gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -512,6 +518,36 @@ static void checkValidFloatingPoint(double value) {
   }
 
   /**
+   * This method is used to get the {@link TypeAdapter} for the specified field. The type
+   * adapter needs to be configured by {@link GsonBuilder#registerFieldAdapterFactory}.
+   *
+   * @param field the field needs to be investigated
+   * @return the {@link TypeAdapter} for the given {@code field}
+   * @since 2.8.1
+   */
+  public <T> TypeAdapter<T> getFieldAdapter(Field field) {
+    if (fieldAdapterFactories == null || fieldAdapterFactories.isEmpty()) {
+      return null;
+    }
+    for (Annotation fieldAnnotation : field.getAnnotations()) {
+      Class<? extends Annotation> annotationType = fieldAnnotation.annotationType();
+      if (fieldAdapterFactories.containsKey(annotationType)) {
```

Comment:
might be more efficient to loop over the factories and call `field.isAnnotationPresent()` rather than looping over the field’s annotations. Only because `isAnnotationPresent()` doesn’t allocate, but `getAnnotations()` needs to build instances of all of the field’s annotations.